GOSSIP.md

```diff
@@ -2,77 +2,323 @@
+the set of peer's IP/port the attacker should control to fill enough of the list
+for the attack to be statistically successful. In addition, we randomize the
+peer eviction to prevent an attacker from predicting it and replace all good
+peers for their own compromised ones.
+
+This is achieved by first categorizing peers in two groups:
+
+1. The unverified pool contains all the peers received through gossip.
+It ensures no Byzantine node can fill it completely by limiting the subset
+of the pool it can affect; it prevents attackers from predicting the set of
+IP/port they must use to successfully eclipse the node.
+
+2. The verified pool contains the peers the node connected to explicitly after
+passing through the filter of the unverified pool. It randomizes the peer
+distribution to the eye of attackers, preventing them from predicting the
+eviction algorithm.
+
+Peers are considered verified when the node has been able to connect to
+them using the Noise protocol.
+
+Both groups clusterize peers into multiple buckets; they select which one
+an added peer belongs to based on a combination of the address of the node the
+information is coming from, the address of the peer to be added and a secret
+generated by the node for the purpose of randomizing the selection process.
+See [Unverified Pool](#unverified-pool) and [Verified Pool](#verified-pool) for
+more details on how the peers are added to the pools.
+
+The peers received from configuration are added directly to the verified pool
+and are marked as "trusted", meaning that they will never get downgraded to the
+unverified pool even if the node cannot connect to them.
+
+### Peer Groups
+
+Peers are grouped by the 16 most significant bits of a peer IP (\16 mask).
+This group is used when selecting a bucket and establishing the node outbound
+connections to prevent nodes from being connected only to local nodes; this
+make block forwarding more effective. NOTE: If all the nodes are in the same
+group, they may not reach their maximum number of outbound connection and this
+should probably be disabled.
+
+### Connections
+
+There is two pool of connections, inbound connection and outbound connections.
+
+Both inbound and outbound connections are used for mempool, gossip and sync
+protocol; but only outbound connections are used for relaying new blocks to
+reduce the network load.
+
+There is a hard limit on the maximum number of outbound connection
+(default to 10).
+
+There is a soft limit on the maximum of inbound connections (default to 100).
+When it is reached, the node will still accept inbound connections, but they will
+be closed right after responding to the first ping. This is a soft limit to allow
+nodes to join the cluster even if the trusted nodes inbound limit has been
+reached. In addition, this prevent a node, in particular a trusted node, to
+have so much inbound connections that it cannot reach its number of outbound
```

Comment:
many?

Reply:
What do you mean?  Replace `so much inbound connection` to `many inbound connections`? I feel this doesn't mean the same, but I may be wrong.

Reply:
»In addition, this prevents a node, in particular a trusted one, to end up with too many inbound connections. This could prevent it from establishing further outbound connections, due to the global maximum«

This is what you mean, I assume?

Reply:
This is not really the issue. The problem is that because all the nodes connect to the trusted nodes, it is already connected to all pooled peers so it will not connect to any. Another option would be to force outbound connection even though a node already have an inbound connection for the peer but this doesn't sound like a good idea.